### PR TITLE
Simon/issue#97/breadcrumbs

### LINKED ISSUE #97: Breadcrumbs
From user interview, 

We would like a button, appearing at the extreme right of the message window, that returns the user to the previous page.  This is useful for learning the pageset, particularly in a group: 

* "How did you get there?" 
* *presses back* 
* "Oh, it's a type of sport" 

The user's mental model for this is a stack - pages that are reached by using 'normal' jump-to-page actions are added to the stack, the back button pops entries from the stack. 

There is a potential optimisation - which is that the back button takes the shortest path back from that page in the history to the root.  I think that's an unexpected behaviour, but the THING to do is to ask the user: 

For the back button - what do you think the back button should do if a user has gone round in a loop a lot for fun? 

Assuming the user has gone to the following pages: 
Root-->food-->flavours-->food-->flavours-->food-->flavours-->food-->flavours-->food-->flavours-->food-->flavours-->food and is now on 'food'.

Should the back button: 

- Unwind the loop repeatedly (jump to every link in the chain)
- Unwind the loop once and then take a shortcut out (jump to flavours then food then root) 
- Immediately take the shortest rout back (jump directly to root from food)

### PR DESCRIPTION
**Pull request by psi77**
Added back button (configurable in same way as other buttons)
Added two stack behaviours;
* full stack which just keeps track of everywhere the user has been
* optimised stack which will roll the stack back to a page earlier in the stack if it is revisited

Currently the optimised stack is the default.
There is no way of the user choosing a preferred stack behaviour but we could add this if it is desired

Closes #97

### CODE REVIEW

File: src/app/board.service.ts [hunk 1]
```diff
@@ -49,6 +53,14 @@ export class BoardService {
     this.currentBoardKey = boardKey;
     const board = this.boardSet.getBoard(this.currentBoardKey);
     this.observer.next(board);
+    this.pageStackService.addPage(boardKey);
+  }
+
+  back() {
+    const boardKey = this.pageStackService.back();
```
Comment (blaise17) [inline src/app/board.service.ts]: This probably shouldn't be called back as it isn't an action, maybe get_prev_page or something similar?

**Comment by psi77** [comment]
Lol

**Comment by joereddington** [comment]
I will have a check of this shortly, heading out to get Nova now ish...

**Comment by joereddington** [comment]
Have had a play.  Works for me.  Awesome work. Only comments would be: 

- Would be good if the button grey'd out at the bottom of the stack
- There's _something_ bothering me about the language in the config: "back button always shown" as opposed to, say 'enable back button'.  I'm unable to put words around it properly yet...

**Pull request merged**